src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,8 +1,64 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
 
-    public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+    public String getSalaryInfo(String[] names, String[] data,
+                                       String dateFrom, String dateTo) {
+        DateTimeFormatter dateTimeFormatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+        LocalDate dateFromWithFormatter = LocalDate.parse(dateFrom, dateTimeFormatter);
+        LocalDate dateToWithFormatter = LocalDate.parse(dateTo, dateTimeFormatter);
+        StringBuilder salaryInfo = new StringBuilder()
+                .append("Report for period ")
+                .append(dateFrom).append(" - ")
+                .append(dateTo);
+        String dataSeparator = " ";
+        StringBuilder date = new StringBuilder();
+        StringBuilder name = new StringBuilder();
+        StringBuilder hour = new StringBuilder();
+        StringBuilder salaryPerHour = new StringBuilder();
+        int earnedPerName = 0;
+        for (int i = 0; i < names.length; i++) {
+            for (int j = 0; j < data.length; j++) {
+                date.delete(0, date.length());
```

Comment:
you can split each line by ` `, receive an array, and address to line parts by indices 
just make these indices (0, 1, 2, 3) constants